tests/nn/moe/test_moe_layer.py

```diff
@@ -3,53 +3,59 @@
 # This source code is licensed under the BSD license found in the
 # LICENSE file in the root directory of this source tree.
 
-import os
+import functools
 import tempfile
 
 import pytest
 import torch
 import torch.distributed as dist
+import torch.multiprocessing as mp
 
 from fairscale.nn import MOELayer, Top2Gate
+from fairscale.utils.testing import torch_version
 
-skip_if_no_cuda = pytest.mark.skipif(not torch.cuda.is_available(), reason="cuda required")
+pytestmark = pytest.mark.skipif(
+    not (torch.cuda.is_available() and torch_version() >= (1, 8, 0)), reason="cuda and torch>=1.8.0 required"
+)
 
-BACKEND = dist.Backend.NCCL if torch.cuda.is_available() else dist.Backend.GLOO  # type: ignore
+devices = ["cuda"]
```

Comment:
why don't we want cpu anymore?

Reply:
i just read the linked issue. Guessing CPU was a stop gap till all_to_all was supported?

Reply:
NCCL does not support CPU tensors.